src/gui_common/PatchNotesDisplayer.cs

```diff
@@ -0,0 +1,170 @@
+﻿using System.Linq;
+using Godot;
+
+/// <summary>
+///   Handles displaying the patch notes from the previous version to the current one
+/// </summary>
+public class PatchNotesDisplayer : VBoxContainer
+{
+    [Export]
+    public bool InsideDialogStyle;
+
+    [Export]
+    public NodePath? PatchNotesPath;
+
+    [Export]
+    public NodePath TitlePath = null!;
+
+    [Export]
+    public NodePath NewVersionsCountLabelPath = null!;
+
+    [Export]
+    public NodePath ViewAllButtonPath = null!;
+
+    [Export]
+    public NodePath ViewAllButtonOutsideScrollPath = null!;
+
+#pragma warning disable CA2213
+    private PatchNotesList patchNotes = null!;
+
+    private Label title = null!;
+    private Label newVersionsCountLabel = null!;
+
+    private Button viewAllButton = null!;
+    private Button viewAllButtonOutsideScroll = null!;
+#pragma warning restore CA2213
+
+    public override void _Ready()
+    {
+        patchNotes = GetNode<PatchNotesList>(PatchNotesPath);
+        title = GetNode<Label>(TitlePath);
+        newVersionsCountLabel = GetNode<Label>(NewVersionsCountLabelPath);
+        viewAllButton = GetNode<Button>(ViewAllButtonPath);
+        viewAllButtonOutsideScroll = GetNode<Button>(ViewAllButtonOutsideScrollPath);
+
+        patchNotes.Visible = false;
+        newVersionsCountLabel.Visible = false;
+
+        if (InsideDialogStyle)
+        {
+            title.Visible = false;
+            patchNotes.StyleWithBackground = false;
+
+            viewAllButton.Visible = false;
+            viewAllButtonOutsideScroll.Visible = true;
+        }
+    }
+
+    public bool ShowIfNewPatchNotesExist()
+    {
+        var lastPlayed = LastPlayedVersion.LastPlayed;
+        if (lastPlayed == null)
+        {
+            // If no known last played version, this is the firs time Thrive is played and there's no need to show
+            // patch notes
+            return false;
+        }
+
+        var currentVersion = Constants.Version;
+
+        // If we aren't playing a newer version then also skip showing
+        if (VersionUtils.Compare(currentVersion, lastPlayed) <= 0)
```

Comment:
If I add some invalid data into `last_played_version.txt` (To simulate that it is corrupt), Thrive will print a *LOT* of exceptions.

Reply:
I'll put in data validation to LastPlayedVersion loading code.